jmix-ui/ui/src/main/java/io/jmix/ui/widget/JmixGroupTable.java

```diff
@@ -567,6 +582,129 @@ protected void groupBy(Object[] properties, boolean rerender) {
             refreshRenderedCells();
             markAsDirty();
         }
+        if (sortOnGroupEnabled) {
+            sortByGroupingProperties(properties);
+        }
+    }
+
+    protected void sortByGroupingProperties(Object[] groupingPropertyIds) {
+        Map<Object, SortDetails> newPropertyIdsToSort = new LinkedHashMap<>();
+        if (isSortEnabled()) {
```

Comment:
try to avoid extra code nesting. The following code can be changed to 

```
if (!isSortEnabled()) {
   return;
}

Map<Object, SortDetails> newPropertyIdsToSort = new LinkedHashMap<>();
// do the stuff
```